### PR TITLE
Format viewpoint test annotations

### LINKED ISSUE #873: Add Test qualifiers to type annotation formatting
Find a way to add viewpoint test checker annotation to format

_Originally posted by @wmdietl in https://github.com/eisop/checker-framework/pull/827#discussion_r1722429879_

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Fix #873

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -300,7 +300,7 @@ allprojects {
 
                 googleJavaFormat().aosp()
                 importOrder('com', 'jdk', 'lib', 'lombok', 'org', 'java', 'javax')
-                formatAnnotations().addTypeAnnotation("PolyInitialized")
+                formatAnnotations().addTypeAnnotation("PolyInitialized").addTypeAnnotation("A").addTypeAnnotation("B").addTypeAnnotation("Bottom").addTypeAnnotation("PolyVP").addTypeAnnotation("ReceiverDependentQual").addTypeAnnotation("Top")
```
Comment (wmdietl) [inline build.gradle:303]: Can you turn this into two arrays: 1) regular type qualifiers that are not in spotless (yet) 2) test type qualifiers that never make sense to add to spotless.
For the second set, can you add a comment/script, that re-creates the array for future additions?
Then, turn the `addTypeAnnotation` into a functional style that passes each array element.

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for looking into this!

**Comment by Ao-senXiong** [comment]
Looks like spotless does not allow `addTypeAnnotation` be used as a functional style.
The following code will throw this error, single annotation array is fine and multiple annotations array will cause this problem:
```
// Regular type qualifiers array that are not in Spotless yet
def regularTypeQualifiers = [
        "PolyInitialized"
]

// Test type qualifiers array
def testTypeQualifiers = [
        "PolyVP", "ReceiverDependentQual"
]

def allQualifiers = regularTypeQualifiers + testTypeQualifiers

// Add all type qualifiers to Spotless
allQualifiers.each { qualifier ->
    formatAnnotations().addTypeAnnotation(qualifier)
}
```
```
* What went wrong:
Could not create task ':checker:spotlessJavaCheck'.
Could not create task ':checker:spotlessJava'.
Multiple steps with name 'No line break between type annotation and type' for spotless format 'java'
```

However, I also notice spotless already support a few of our testannotations so I can simplify it.
https://github.com/diffplug/spotless/blob/85a0bebc71dcf59f56f0ac31b8b2507294cff78d/lib/src/main/java/com/diffplug/spotless/java/FormatAnnotationsStep.java#L51

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**